src/tests/AabbTest.cpp

```diff
@@ -107,6 +113,97 @@ void GeoTest::aabb() {
   }
 }
+
+void GeoTest::obbConstruction() {
+  OBB obb1;
+  const vec3f center(0, 0, 0);
+  const vec3f dimensions(20, 2, 10);
+  const quatf rot1 = quatf::FromTwoVectors(vec3f::UnitY(), vec3f::UnitZ());
+  OBB obb2(center, dimensions, rot1);
+
+  CORRADE_VERIFY(obb2.center().isApprox(center));
+  CORRADE_VERIFY(obb2.sizes().isApprox(dimensions));
+  CORRADE_VERIFY(obb2.halfExtents().isApprox(dimensions / 2));
+  CORRADE_VERIFY(obb2.rotation().coeffs().isApprox(rot1.coeffs()));
+
+  box3f aabb(vec3f(-1, -2, -3), vec3f(3, 2, 1));
+  OBB obb3(aabb);
+  CORRADE_VERIFY(obb3.center().isApprox(aabb.center()));
+  CORRADE_VERIFY(obb3.toAABB().isApprox(aabb));
+}
+
+void GeoTest::obbFunctions() {
+  const vec3f center(0, 0, 0);
+  const vec3f dimensions(20, 2, 10);
+  const quatf rot1 = quatf::FromTwoVectors(vec3f::UnitY(), vec3f::UnitZ());
+  OBB obb2(center, dimensions, rot1);
+  CORRADE_VERIFY(obb2.contains(vec3f(0, 0, 0)));
+  CORRADE_VERIFY(obb2.contains(vec3f(-5, -2, 0.5)));
+  CORRADE_VERIFY(obb2.contains(vec3f(5, 0, -0.5)));
+  CORRADE_VERIFY(!obb2.contains(vec3f(5, 0, 2)));
+  CORRADE_VERIFY(!obb2.contains(vec3f(-10, 0.5, -2)));
+
+  const box3f aabb = obb2.toAABB();
+  CORRADE_COMPARE_AS(aabb.min().x(), -10, float);
+  CORRADE_COMPARE_AS(aabb.min().y(), -5, float);
+  CORRADE_COMPARE_AS(aabb.min().z(), -1, float);
+  CORRADE_COMPARE_AS(aabb.max().x(), 10, float);
+  CORRADE_COMPARE_AS(aabb.max().y(), 5, float);
+  CORRADE_COMPARE_AS(aabb.max().z(), 1, float);
+
+  const Transform identity = obb2.worldToLocal() * obb2.localToWorld();
+  CORRADE_VERIFY(identity.isApprox(Transform::Identity()));
+  CORRADE_VERIFY(obb2.contains(obb2.localToWorld() * vec3f(0, 0, 0)));
+  CORRADE_VERIFY(obb2.contains(obb2.localToWorld() * vec3f(1, -1, 1)));
+  CORRADE_VERIFY(obb2.contains(obb2.localToWorld() * vec3f(-1, -1, -1)));
+  CORRADE_VERIFY(!obb2.contains(obb2.localToWorld() * vec3f(-1, -1.5, -1)));
+  CORRADE_COMPARE_AS(obb2.distance(vec3f(-20, 0, 0)), 10, float);
+  CORRADE_COMPARE_AS(obb2.distance(vec3f(-10, -5, 2)), 1, float);
+}
+
+void GeoTest::coordinateFrame() {
+  /*
+    const vec3f origin(1, -2, 3);
+    const vec3f up(0, 0, 1);
+    const vec3f front(-1, 0, 0);
+    quatf rotation = quatf::FromTwoVectors(ESP_UP, up) *
+                     quatf::FromTwoVectors(ESP_FRONT, front);
+    Transform xform;
+    xform.rotate(rotation);
+    xform.translate(origin);
+
+    CoordinateFrame c1(up, front, origin);
+    CORRADE_VERIFY(c1.up().isApprox(up));
+    CORRADE_VERIFY(c1.gravity().isApprox(-up));
+    CORRADE_VERIFY(c1.front().isApprox(front));
+    CORRADE_VERIFY(c1.back().isApprox(-front));
+    CORRADE_VERIFY(c1.up().isApprox(rotation * ESP_UP));
+    CORRADE_VERIFY(c1.front().isApprox(rotation * ESP_FRONT));
+    CORRADE_VERIFY(c1.origin().isApprox(origin));
+    CORRADE_VERIFY(c1.rotationWorldToFrame().isApprox(rotation));
+
+    CoordinateFrame c2(rotation, origin);
+    CORRADE_COMPARE(c1, c2);
```

Comment:
@mosra: Your doc says item must be printable via Debug, which is NOT the case here. How can I fix this problem at this point?

Reply:
Part of the error message reads:
`habitat-sim/src/deps/corrade/src/Corrade/TestSuite/Comparator.h:270:7: error: invalid operands to binar
y expression ('Corrade::Utility::Debug' and 'const std::__1::basic_string<char>')
      << *actualValue << Utility::Debug::newline << "        but expected\n       " << *expectedValue;
      ^  ~~~~~~~~~~~~
/Users/user/habitat-sim/src/deps/corrade/src/Corrade/TestSuite/Comparator.h:326:16: note: in instantiation of membe
r function 'Corrade::TestSuite::Comparator<std::__1::basic_string<char> >::printMessage' requested here
    comparator.printMessage(flags, out, actual, expected);`

Reply:
[`#include <Corrade/Utility/DebugStl.h>`](https://doc.magnum.graphics/corrade/classCorrade_1_1Utility_1_1Debug.html#Utility-Debug-stl)

Reply:
For the actual `CoordinateFrame` comparison above, unless you'd want to write an `operator<<(Mn::Debug&, const CoordinateFrame&)` for it (which might be only useful for this particular test and nowhere else), the easiest would be to do just

```suggestion
    CORRADE_VERIFY(c1 == c2);
```

That way you don't get any useful diagnostic when the two aren't equal, but you didn't get that with gtest either.